crypto/crypto.go

```diff
@@ -66,31 +62,29 @@ func trimLeftZeros(b []byte) []byte {
 }
 
 // ValidateSignatureValues checks if the signature values are correct
-func ValidateSignatureValues(v byte, r, s *big.Int) bool {
-	// TODO: ECDSA malleability
+func ValidateSignatureValues(v byte, r, s *big.Int, isHomestead bool) bool {
+	// r & s must not be nil
 	if r == nil || s == nil {
 		return false
 	}
 
-	if v > 1 {
+	// r & s must be positive integer
+	if r.Cmp(one) < 0 || s.Cmp(one) < 0 {
 		return false
 	}
 
-	rr := r.Bytes()
-	rr = trimLeftZeros(rr)
-
-	if bytes.Compare(rr, secp256k1N) >= 0 || bytes.Compare(rr, one) < 0 {
+	// v must be 0 or 1
+	if v > 1 {
 		return false
 	}
 
-	ss := s.Bytes()
-	ss = trimLeftZeros(ss)
-
-	if bytes.Compare(ss, secp256k1N) >= 0 || bytes.Compare(ss, one) < 0 {
+	// From Homestead, s must be less than secp256k1n/2
```

Comment:
Please double check me, but if we negate invalid condition for `s` validation, we are getting `s <= secp256k1n/2`?

```suggestion
        // From Homestead, s must be less or equal than secp256k1n/2
```

Reply:
Fixed 5c0da0216cb78fe695c7019347e5be3594afb623 @Stefan-Ethernal